pkg/shell/util.tsx

```diff
@@ -101,7 +111,8 @@ export class CompiledComponents {
 
     load(section: string): void {
         Object.entries(this.manifests).forEach(([name, manifest]) => {
-            Object.entries(manifest[section] || { }).forEach(([prop, info]) => {
+            const manifest_section = (manifest[section] || {}) as ManifestSection;
+            Object.entries(manifest_section).forEach(([prop, info]) => {
```

Comment:
> This cast is weird and IMHO detrimental.

Yes... it is keeping the status quo, but we should try to do better eventually, agreed.

> I.e. you'll get runtime errors because you forgot to check the type. That's the opposite of what typing should achieve.

Hmm, I would call this input validation rather than static typing. Currently, we validate manifest input by just pretending it's all valid and let the shell crash when it isn't. (And we have been fine with that since forever.) Static typing makes this obvious, but by itself is not enough to improve it.

We would have to write new code to do real input validation. I was hoping there is a way to get TypeScript to do that. Here are some good pointers: https://stackoverflow.com/questions/33800497/check-if-an-object-implements-an-interface-at-runtime-with-typescript